compose/config/config.py

```diff
@@ -668,7 +734,13 @@ def is_url(build_path):
 
 def validate_paths(service_dict):
     if 'build' in service_dict:
-        build_path = service_dict['build']
+        build = service_dict.get('build', {})
+
+        if isinstance(build, six.string_types):
+            build_path = build
+        elif isinstance(build, dict) and 'context' in build:
+            build_path = build['context']
```

Comment:
The issue is that if we normalize `build` in `process_service` then v1 schema validation will fail. We could potentially add a `normalize_build()` immediately after `validate_against_service_schema`, but then we still need conditional logic in `validation.py` that you also commented on.